types/es-abstract/2018/IsStringPrefix.d.ts

```diff
@@ -0,0 +1,2 @@
+declare function IsStringPrefix(p: string, q: string): boolean;
```

Comment:
is there a way in typescript to do something like `p extends q` as a predicate, so that `'f'` is seen as a prefix of `'foo'`?

Reply:
**TypeScript** doesn’t currently support this.